### PR TITLE
fix(StatusQ): Fixing a crash when the source model is a nim model

### PR DESCRIPTION
**Pull request by alexjba**
#### Needed for: https://github.com/status-im/status-desktop/pull/12160

Nim models are crashing when calling `itemData`. A workaround for this is to re-implement the `itemData` calls in the writableModel.

Moreover, the default `itemData` implementation in the QAbstractItemModel seems to process only the static roles.

### Affected areas

WritableProxyModel

### CODE REVIEW

**Review by alexandraB99 — APPROVED** [review]
LGTM

File: ui/StatusQ/src/writableproxymodel.cpp [hunk 1]
```diff
@@ -671,19 +671,27 @@ QMap<int, QVariant> WritableProxyModel::itemData(const QModelIndex& index) const
     if (!index.isValid())
         return {};
 
-    return QAbstractProxyModel::itemData(index);
+    QMap<int, QVariant> result;
+    
+    for (auto& role : roleNames().keys())
```
Comment (caybro) [inline ui/StatusQ/src/writableproxymodel.cpp:676]: https://github.com/KDE/clazy/blob/1.11/docs/checks/README-container-anti-pattern.md

**Review by caybro — APPROVED** [review]
Otherwise LGTM

File: ui/StatusQ/src/writableproxymodel.cpp (hunk 1 above)
Comment (alexjba) [reply ui/StatusQ/src/writableproxymodel.cpp:676]: ohh yes!

**Pull request merged**